hack/e2e.go

```diff
@@ -207,6 +207,11 @@ func run(deploy deployer) error {
 	}
 
 	if *up {
+		// If we tried to bring the Kubemark cluster up, make a courtesy
+		// attempt to bring it down so we're not leaving resources around.
+		if *down {
+			defer xmlWrap("TearDown", deploy.Down)
+		}
 		// Start the cluster using this version.
 		if err := xmlWrap("Up", deploy.Up); err != nil {
 			return fmt.Errorf("starting e2e cluster: %s", err)
```

Comment:
I assure you this was not my intention, and I would like to see it working again soon too -- ideally with the lowest ongoing maintenance burden.

It is much easier to reason about control flow in Golang than bash. It seems the easiest way to understand the test output is if we do the following (for both cluster and kubemark down):
* Ensure we always attempt to teardown if we attempt to setup.
* Attempt to teardown exactly once after setting up.

What are your thoughts about this flow? Is there a good way to achieve it?